airbyte-workers/src/main/java/io/airbyte/workers/normalization/NormalizationAirbyteStreamFactory.java

```diff
@@ -0,0 +1,115 @@
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.workers.internal.AirbyteStreamFactory;
+import java.io.BufferedReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Creates a stream from an input stream. The produced stream attempts to parse each line of the
+ * InputStream into a AirbyteMessage. If the line cannot be parsed into a AirbyteMessage it is
+ * assumed to be from dbt. dbt [error] messages are also parsed
+ *
+ * <p>
+ * If a line starts with a AirbyteMessage and then has other characters after it, that
+ * AirbyteMessage will still be parsed. If there are multiple AirbyteMessage records on the same
+ * line, only the first will be parsed.
+ */
+public class NormalizationAirbyteStreamFactory implements AirbyteStreamFactory {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(NormalizationAirbyteStreamFactory.class);
+
+  private final MdcScope.Builder containerLogMdcBuilder;
+  private final Logger logger;
+  private final List<String> dbtErrors = new ArrayList<>();
+
+  public NormalizationAirbyteStreamFactory(final MdcScope.Builder containerLogMdcBuilder) {
+    this(LOGGER, containerLogMdcBuilder);
+  }
+
+  NormalizationAirbyteStreamFactory(final Logger logger, final MdcScope.Builder containerLogMdcBuilder) {
+    this.logger = logger;
+    this.containerLogMdcBuilder = containerLogMdcBuilder;
+  }
+
+  @Override
+  public Stream<AirbyteMessage> create(final BufferedReader bufferedReader) {
+    return bufferedReader
+        .lines()
+        .flatMap(line -> {
+          final Optional<JsonNode> jsonLine = Jsons.tryDeserialize(line);
+          if (jsonLine.isEmpty()) {
+            // we log as info all the lines that are not valid json.
+            try (final var mdcScope = containerLogMdcBuilder.build()) {
+              logger.info(line);
+              // this is really hacky and vulnerable to picking up lines we don't want,
+              // however it is only for destinations that are using dbt version < 1.0.
+              // For v1 + we switch on JSON logging and parse those in the next block.
+              if (line.contains("[error]")) {
+                dbtErrors.add(line);
+              }
+            }
+          }
+          return jsonLine.stream();
+        })
+        .flatMap(jsonLine -> {
+          final Optional<AirbyteMessage> m = Jsons.tryObject(jsonLine, AirbyteMessage.class);
+          if (m.isEmpty()) {
+            // valid JSON but not an AirbyteMessage, so we assume this is a dbt json log
+            try {
+              final String logLevel = jsonLine.get("level").isNull() ? "" : jsonLine.get("level").asText();
+              final String logMsg = jsonLine.get("msg").isNull() ? "" : jsonLine.get("msg").asText();
+              try (final var mdcScope = containerLogMdcBuilder.build()) {
+                switch (logLevel) {
```

Comment:
This seems like a required field from dbt and any other json line _should_ be an AirbyteMessage, however I've added a case to send the line to log info just in case.